### PR TITLE
Feature/wifi 917soc wifisdk matter support

### PR DESCRIPTION
**Pull request by sharad-patil24**
#### Problem / Feature
Added support files for Si917 SOC

### CODE REVIEW

**Review by mkardous-silabs — CHANGES_REQUESTED** [review]
Can you Remove all :

- [x] extension specific configuration (files and configurations)
- [ ] zap-generated files
- [ ] non-necessary files
- [ ] Make sure restyler passes

File: matter/si91x/siwx917/BRD4325B/config/em_core_generic.h [hunk 1]
```diff
@@ -0,0 +1,167 @@
+/***************************************************************************/ /**
```
Comment (jmartinez-silabs) [inline matter/si91x/siwx917/BRD4325B/config/em_core_generic.h:1]: nit: removing the `/  /` at the end of the line here and in some function briefs of this file and `emlib_core_debug_config.h` and then applying the styler should fix the wrong indentation done by the styler.

**Review by jmartinez-silabs — APPROVED** [review]

File: matter/si91x/siwx917/BRD4325B/config/em_core_generic.h (hunk 1 above)
Comment (sharad-patil24) [reply matter/si91x/siwx917/BRD4325B/config/em_core_generic.h]: done

File: matter/si91x/siwx917/BRD4325B/autogen/sl_component_catalog.h [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+#define SL_CATALOG_MATTER_ON_OFF_PRESENT
+#define SL_CATALOG_MATTER_OPERATIONAL_CREDENTIALS_PRESENT
+#define SL_CATALOG_MATTER_OTA_REQUESTOR_PRESENT
+#define SL_CATALOG_MATTER_SOFTWARE_DIAGNOSTICS_PRESENT
+#define SL_CATALOG_MATTER_SWITCH_PRESENT
+#define SL_CATALOG_MATTER_THREAD_NETWORK_DIAGNOSTICS_PRESENT
+#define SL_CATALOG_MATTER_TIME_FORMAT_LOCALIZATION_PRESENT
+#define SL_CATALOG_MATTER_USER_LABEL_PRESENT
+#define SL_CATALOG_MATTER_BLE_PRESENT
+#define SL_CATALOG_MATTER_WIFI_NETWORK_DIAGNOSTICS_PRESENT
```
Comment (mkardous-silabs) [inline matter/si91x/siwx917/BRD4325B/autogen/sl_component_catalog.h:31]: All these defines are not necessary for CSA.

**Review by mkardous-silabs — CHANGES_REQUESTED** [review]

File: matter/si91x/siwx917/BRD4325B/autogen/sl_component_catalog.h (hunk 2 above)
Comment (sharad-patil24) [reply matter/si91x/siwx917/BRD4325B/autogen/sl_component_catalog.h]: Removed unnecessary MACROS

**Review by mkardous-silabs — APPROVED** [review]

**Pull request merged**